Xamarin.PropertyEditing/Common/FilePath.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+namespace Xamarin.PropertyEditing
+{
+	public class FilePath
+	{
+		public string Source { get; set; }
+		public bool IsDirectory { get; set; }
```

Comment:
You can't actually depend on something from the value to determine the behavior of the control as you may not have a value to begin with. File and Directory path need to be separate types to distinguish behaviors. The base of the controls can largely share behavior with a couple overrides to define file or directory specifics.

Reply:
done.